### PR TITLE
NRI raider ship maintenance PR

### PR DESCRIPTION
**Pull request by Stalkeros2**
## About The Pull Request
This PR is meant to fix a bunch of rather major issues the NRI raiders had. I, er, think this is it huh.
One of them was and is the general lack of roleplay incentives for both the raiders and the station to cooperate. Mostly that was caused by the raiders not having any reason to go peacefully. Or rather, each side not having any reason to go peacefully - there was nothing that could be used as an argument, and the unnamed "violated regulation" did not help with that either. Thus, I am adding a randomly generated regulatory violations. They can be both "right" (= mostly correct) and "wrong" (= mostly incorrect) and can be used to let NRI raiders leave the station alone without paying anything.
image

Thanks to @dawsonkeyes, the shuttle has been **completely** remapped. And is now actually *kinda* in line with the other pirate shuttles.
(Their announcement console also works yeah, and also drops NOTHING on destruction so people can't reconstruct nor deconstruct it to make themselves the NT one.)
image
image
image

## How This Contributes To The Skyrat Roleplay Experience
Antags that can actually roleplay with the station and not turn it into a mess is **very freaking good**.
Fixing issues according to the received sparse feedback.

## Changelog

:cl: Stalkeros, Lukaster42
add: NRI Raiders, now (hopefully) more roleplay-oriented.
/:cl:

### CODE REVIEW

File: code/modules/events/ghost_role/pirates.dm [hunk 1]
```diff
@@ -97,9 +97,31 @@
...
 			ship_template = /datum/map_template/shuttle/pirate/imperial_enclave
+			payoff = 20000
+			var/number = rand(1,999)
+			///Station name one is the most important pick and is pretty much the station's main argument against getting fined, thus it better be mostly always right.
+			var/station_designation = pick_weight(list("Nanotrasen Research Station" = 70,
+												"Nanotrasen Refueling Outpost" = 5,
+												"Interdyne Pharmaceuticals Chemical Factory" = 5,
+												"Free Teshari League Engineering Station" = 5,
+												"Agurkrral Military Base" = 5,
+												"Sol Federation Interceptor" = 5,
```
Comment (ghost) [inline code/modules/events/ghost_role/pirates.dm]: Threat.content has the text 'your station', Interceptor is a kind of ship :b While the report's meant to be occasionally wrong, it shouldn't be contradictory

Comment (Stalkeros2) [reply code/modules/events/ghost_role/pirates.dm]: I assume that was fixed.

**Review by Paxilmaniac — CHANGES_REQUESTED** [review]
Map vibe check:

image
Try to not use double thick walls, extend or move the medbay a bit, whatever you have to do.

image
I feel that both the brig and medical areas could use more decals to them, try a trim line going around the walls in red and blue respective to each area.

image
I'm fairly certain the medpen refiller needs a link to plumbing machines to function, so this will be useless.

image
Remove that center RTG there so the ones on either side of it can actually be accessed.

image
Remove the black box, this is loot that's literally only useful for traitors and there should only be one of them per round.

image
Bedsheets on top of the cryopods looks weird, especially since they are facing the wrong way, just remove them from both here and the captain's office.

image
Make sure to include both a coffee pot and coffee carts so they can actually use this.

image
I'd prefer you just used normal secured lockers, actually secure crates might be better here, your pick.

image
Two other little things about the bridge, it seems like an odd choice to have green carpet here of all places, and also have enough seating for all of the crew to sit in the bridge.

image
I too enjoy sitting on the couch that stares into a wall.

image
Some of the random gear around the ship can be called just slightly overtuned, not even related to combat gear, pirates don't need an advanced mop.

image
Pirates don't need a rapid pipe dispenser.

image
If that's a medal of captaincy I ask you either remove it or replace it with another medal type as this is a traitor objective item that should only be spawning in set on station locations.

image
Another nitpick but disablers don't seem fitting to the whole theme of the NRI, just me though you do what you want.

image
I don't think the atmos room really needs to be this large for, again, a pirate ship.

In fact one of the bigger issues aside from the ones I've mentioned is that this ship is just way too large, and definitely wasn't meant to be used for this purpose. I know this because I made the original version, this is just shuttlestation turned into a pirate vessel.

**Review by Paxilmaniac — APPROVED** [review]
Map is good enough now. That's a whole lot of lines of code, too bad I ain't reading them.

**Comment by Lukaster42** [comment]
One last commit remains.

**Comment by Stalkeros2** [comment]
no

**Comment by Stalkeros2** [comment]
lol

File: code/modules/events/ghost_role/pirates.dm [hunk 2]
```diff
@@ -97,9 +97,31 @@
...
+			payoff = 20000
+			var/number = rand(1,99)
+			///Station name one is the most important pick and is pretty much the station's main argument against getting fined, thus it better be mostly always right.
+			var/station_designation = pick_weight(list("Nanotrasen Research Station" = 70,
+												"Nanotrasen Refueling Outpost" = 5,
+												"Interdyne Pharmaceuticals Chemical Factory" = 5,
+												"Free Teshari League Engineering Station" = 5,
+												"Agurkrral Military Base" = 5,
+												"Sol Federation Embassy" = 5,
+												"Novaya Rossiyskaya Imperiya Civilian Port" = 5,))
```
Comment (Tastyfish) [inline code/modules/events/ghost_role/pirates.dm]:
```suggestion
			var/station_designation = pick_weight(list(
				"Nanotrasen Research Station" = 70,
				"Nanotrasen Refueling Outpost" = 5,
				"Interdyne Pharmaceuticals Chemical Factory" = 5,
				"Free Teshari League Engineering Station" = 5,
				"Agurkrral Military Base" = 5,
				"Sol Federation Embassy" = 5,
				"Novaya Rossiyskaya Imperiya Civilian Port" = 5,
			))
```

File: code/modules/events/ghost_role/pirates.dm [hunk 3]
```diff
@@ -97,9 +97,31 @@
...
+			///"right" = Right for the raiders to use as an argument; usually pretty difficult to avoid.
+			var/right_pick = pick("high probability of NRI-affiliated civilian casualties aboard the facility",
+									"highly increased funding by the SolFed authorities; neglected NRI-backed subsidiaries' contracts",
+									"unethical hiring practices and unfair payment allocation for the NRI citizens",
+									"recently discovered BSA-[number] or similar model in close proximity to the neutral space aboard this or nearby affiliated facility")
```
Comment (Tastyfish) [inline code/modules/events/ghost_role/pirates.dm]:
```suggestion
			var/right_pick = pick(
				"high probability of NRI-affiliated civilian casualties aboard the facility",
				"highly increased funding by the SolFed authorities; neglected NRI-backed subsidiaries' contracts",
				"unethical hiring practices and unfair payment allocation for the NRI citizens",
				"recently discovered BSA-[number] or similar model in close proximity to the neutral space aboard this or nearby affiliated facility",
			)
```

File: code/modules/events/ghost_role/pirates.dm [hunk 4]
```diff
@@ -97,9 +97,31 @@
...
+			///"wrong" = Loosely based accusations that can be easily disproven if people think.
+			var/wrong_pick = pick("inadequate support of the local producer",
+									"unregulated production of Gauss weaponry aboard this installation",
+									"SolFed-backed stationary military formation on the surface of Indecipheres",
+									"AUTOMATED REGULATORY VIOLATION DETECTION SYSTEM CRITICAL FAILURE. PLEASE CONTACT AND INFORM THE DISPATCHED AUTHORITIES TO RESOLVE THE ISSUE. \
+									ANY POSSIBLE INDENTURE HAS BEEN CLEARED. WE APOLOGIZE FOR THE INCONVENIENCE")
```
Comment (Tastyfish) [inline code/modules/events/ghost_role/pirates.dm]:
```suggestion
			var/wrong_pick = pick(
				"inadequate support of the local producer",
				"unregulated production of Gauss weaponry aboard this installation",
				"SolFed-backed stationary military formation on the surface of Indecipheres",
				"AUTOMATED REGULATORY VIOLATION DETECTION SYSTEM CRITICAL FAILURE. PLEASE CONTACT AND INFORM THE DISPATCHED AUTHORITIES TO RESOLVE THE ISSUE. \
					ANY POSSIBLE INDENTURE HAS BEEN CLEARED. WE APOLOGIZE FOR THE INCONVENIENCE",
			)
```

File: modular_skyrat/modules/encounters/code/enclave_pirates.dm [hunk 5]
```diff
@@ -124,6 +115,13 @@
 	suffix = "enclave"
 	name = "pirate ship (NRI Enforcer-Class Starship)"
 
+/area/shuttle/pirate/nri
+	name = "NRI Starship"
```
Comment (Tastyfish) [inline modular_skyrat/modules/encounters/code/enclave_pirates.dm]: Is this intentionally still using the "pirate" shuttle ID?

**Review by Tastyfish — CHANGES_REQUESTED** [review]
Mainly fixing some wacky whitespace

File: modular_skyrat/modules/encounters/code/enclave_pirates.dm (hunk 5 above)
Comment (Stalkeros2) [reply modular_skyrat/modules/encounters/code/enclave_pirates.dm]: Yes, or otherwise their objective doesn't show up and register at the end of the round.

**Review by Tastyfish — APPROVED** [review]

File: code/modules/events/ghost_role/pirates.dm [hunk 6]
```diff
@@ -35,9 +36,10 @@
 	typepath = /datum/round_event/pirates/dutchman
 	weight = 0
 
-/datum/round_event_control/pirates/enclave
-	name = "Space Pirates - Imperial Enclave"
-	typepath = /datum/round_event/pirates/enclave
+/datum/round_event_control/pirates/nri
```
Comment (Zonespace27) [inline code/modules/events/ghost_role/pirates.dm:39]: do me a favor and modularize everything between these skyrat edit comments

Comment (Stalkeros2) [reply code/modules/events/ghost_role/pirates.dm:39]: Spent a good while trying to and couldn't. This is horrible.

**Comment by Stalkeros2** [comment]
Should also mention that I'll try and port them as a Stories system event instead at a later date, likely when the aforementioned system gets merged. So I assume that all the current non-modularity can be kinda ignored.

**Comment by Stalkeros2** [comment]
need to fix it properly when i get home

**Review by Paxilmaniac — CHANGES_REQUESTED** [review]
I need you to show me screenshots or even a video of the new ship successfully spawning and moving without leaving any part of it behind and or breaking.

File: modular_skyrat/modules/assault_operatives/code/base_alarm.dm [hunk 7]
```diff
@@ -60,15 +64,16 @@
 		iterating_alarm.triggered = TRUE
 		if(!iterating_alarm.alarm_playing)
 			iterating_alarm.alarm_playing = TRUE
-			playsound(iterating_alarm, 'modular_skyrat/modules/assault_operatives/sound/goldeneyealarm.ogg', 30)
-			addtimer(CALLBACK(iterating_alarm, PROC_REF(alarm_sound)), 65)
+			playsound(iterating_alarm, alarm_sound_file, 30)
+			addtimer(CALLBACK(iterating_alarm, PROC_REF/alarm_sound), alarm_cooldown)
```
Comment (Tastyfish) [inline modular_skyrat/modules/assault_operatives/code/base_alarm.dm]:
```suggestion
			addtimer(CALLBACK(iterating_alarm, PROC_REF(alarm_sound)), alarm_cooldown)
```

**Comment by Stalkeros2** [comment]
Lukaster broke the shuttle so I'll have to fix it tomorrow oh my fucking god.

**Review by Paxilmaniac — APPROVED** [review]
If this breaks ill mail you an armed explosive

**Pull request merged**